### PR TITLE
Create a test suite skeleton

### LINKED ISSUE #23: Add a test suite
## Problem
We are blind to breakages in the example library, such as the one described [here](https://github.com/quil-lang/sbcl-librarian/issues/21#issuecomment-1067995855), that result from API changes/new features.

## Solution
Write some tests that check that the example library builds and runs correctly. Unit tests for individual features are also probably good idea.

### PR DESCRIPTION
**Pull request by kartik-s**
Resolves #23.

## Notes
- **only runs on MacOS for now**, will do Windows in another PR
- you need to have pytest installed to run the Python tests
- see README for instructions on how to write Python tests + add test libraries
- no C tests yet, will add in another PR

## How to run the test suite
```lisp
(ql:quickload "sbcl-librarian/tests")
(asdf:test-system "sbcl-librarian")
```

### Example run
```lisp
kssingh-m:~ kssingh$ sbcl
This is SBCL 2.1.10.11.x86-null-tn.2-711698b4e, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
* (ql:quickload "sbcl-librarian/tests")
To load "sbcl-librarian/tests":
  Load 1 ASDF system:
    sbcl-librarian/tests
; Loading "sbcl-librarian/tests"

("sbcl-librarian/tests")
* (asdf:test-system "sbcl-librarian")
SBCL-LIBRARIAN.TESTS (Suite)
  TEST-BUILD-LIBS.........................................................[ OK ]
  TEST-PYTHON
============================= test session starts ==============================
platform darwin -- Python 3.9.10, pytest-7.1.1, pluggy-1.0.0
rootdir: /Users/kssingh/workspace/sbcl-librarian/tests/python
collected 1 item

test_square.py .                                                         [100%]

============================== 1 passed in 1.00s ===============================
..........................................................................[ OK ]

T
*
```

### CODE REVIEW

**Comment by karlosz** [comment]
Do you mind squashing the commits?

File: tests/libs/libsquare/package.lisp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+(defpackage #:sbcl-librarian/tests/libs/libsquare
```
Comment (kilimanjaro) [inline tests/libs/libsquare/package.lisp:1]: This is totally minor, but I noticed that here you are using slashes in the package name, whereas below (in `tests/package.lisp`) you use dots. Any particular reason for this? One thing we have done for some of the quil projects is uses slashes in ASDF system names (for "secondary systems", this is needed for ASDF to find them) and dots in packages, but I couldn't really tell you why we do that.

**Review by kilimanjaro — COMMENTED** [review]
This is a nice & useful addition.

File: tests/libs/libsquare/package.lisp (hunk 1 above)
Comment (kartik-s) [reply tests/libs/libsquare/package.lisp:1]: No good reason, I pretty much just cargo-culted [this package name](https://github.com/quil-lang/sbcl-librarian/blob/eb084c07d57ec5591b0e8feaa5256d5b38cda8c3/example/package.lisp#L1) from the example. I did try to copy the dot convention used in the Quilc packages for the test package, but I missed it here.

Now that you mention it, this package name should probably just be something like `sbcl-librarian.tests.libsquare`.

Comment (svetlyak40wt) [reply tests/libs/libsquare/package.lisp]: Slashes in package names usually are used in "package inferred" ASDF systems. In such systems package names mirror a file structure and also are names of implicitly created secondary ASDF systems.

**Pull request merged**